refactor: comment system

Youpen-y opened the pull request:
## What does the PR do? (Required)
**Refactor comment system**
> Why it is necessary?
> The `[params.comments]` in the `hugo.toml/config.toml` is misleading to control the Disqus comment system
- [x] Use `[params.comments]` as the global switch to control the comment system 
- [x] Use similar option structure to specify the chosed one from `[params.disqus]/[params.giscus]/[params.utterances]`
- [x] provide page granularity comment system configuration in front matter

Simply put, first check the [params.comments] setting in config.toml.
- If it is false, no comment system will be enabled for any page.
- If it is true, then check the `comments` option in the page's front matter; 
     - if it is false, the comment system will not be shown on that page.
     - if it is omitted, the default value is true.

Additionally, you can choose different comment systems for different posts.
Set `[params.disqus]`, `[params.giscus]`,  and `[params.utterances]` all to false in `config.toml` to disable them globally,
then configure the desired system for each post individually in its front matter.
```text
+++
...
comments = true # this is not necessary, cause it's default value is true
[utterances]
    enable = true
+++
```
`post2` front matter:
```text
+++
...
comments = true
[giscus]]
    enable = true
+++
```
 
## Checklist (Required)

- [x] I have tested the changes on my local machine partly
- [ ] I have added relevant documentation and tests for the changes
- [x] I have followed the style guidelines of this project

## Evidence (Required)
<img width="400" height="300" alt="Screenshot from 2025-09-11 00-04-43" src="https://github.com/user-attachments/assets/cbd382f0-79e8-4bd0-8677-5dd7387eedf9" />
<img width="400" height="300" alt="Screenshot from 2025-09-11 00-05-14" src="https://github.com/user-attachments/assets/973199a8-d3f5-4042-b9fc-04dddc7b452e" />

Cause i don't have Disqus accout, more test need to be checked on Disqus.

2KAbhishek reviewed (changes requested):
Thank you, I like the refactoring proposal, sharing some suggestion to simplify things and bring more clarity:

- Rename the global comment config to `params.global_commnets`
- Rename the page comment config to `params.page_comments` or anything more relevant for the page frontmatter, I am looking for the distinction in names to be clear
- The system config can be determined by a new config option called: `params.comment_system` with options like discus, giscus and utterance - anything specific to these comment system can have its own param section like: params.utterances.repo

Youpen-y commented:
The suggestion is specific, and i'll  try to make it in another free time.

Youpen-y committed:
refactor: comment system

- add global comment system selection (disqus, giscus, utterances)
- support enable/disable comments per page via front matter

Youpen-y committed:
update: example-config.toml
